### PR TITLE
Changes to facilitate aux-variable advection

### PR DESCRIPTION
**Pull request by GiudGiud**
@muelcm I'd recommend you work off of this branch until it gets merged

This is rebased off of next since RC UO PR is not merged yet

### CODE REVIEW

**Comment by lindsayad** [comment]
@loganharbour this coverage change in the framework is just because of the new compilers right? It looks exactly like the line count you showed me

File: modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C [hunk 1]
```diff
@@ -291,6 +321,10 @@ INSFVRhieChowInterpolator::execute()
   if (_a_data_provided)
     return;
 
+  // If advecting with auxiliary variables, no need to try to run those kernels
+  if (_sys.number() != _u->sys().number())
+    return;
```
Comment (lindsayad) [inline modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:326]: I would prefer returning based off whether the `_velocity_interp_method` is average and then just combine with an `||` in the early return above. Then we are not unnecessarily gathering the RC data even when the advecting variables are in the nonlinear system

File: modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C [hunk 2]
```diff
@@ -432,6 +467,9 @@ INSFVRhieChowInterpolator::getVelocity(const Moose::FV::InterpMethod m,
   // Return if Rhie-Chow was not requested
   if (m == Moose::FV::InterpMethod::Average)
     return velocity;
+  mooseAssert((m == _velocity_interp_method) || _a_data_provided,
```
Comment (lindsayad) [inline modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:470]:
```suggestion
  mooseAssert(((m == Moose::FV::InterpMethod::RhieChow) && (_velocity_interp_method == Moose::FV::InterpMethod::RhieChow)) || _a_data_provided,
```

**Review by lindsayad — CHANGES_REQUESTED** [review]
Overall looks good. Sad to see us moving towards supporting average instead of banning it, but I guess it's ok as a temporary solution

File: modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C (hunk 1 above)
Comment (lindsayad) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:326]: Wait never mind ... I understand why you did this ... I actually compute the residuals within `gatherRCData` for some classes

**Review by lindsayad — APPROVED** [review]

**Comment by loganharbour** [comment]
> @loganharbour this coverage change in the framework is just because of the new compilers right? It looks exactly like the line count you showed me

Correct, this is what happens when people rebase on next :-)

**Comment by GiudGiud** [comment]
I did rebase on next to start from the RC PR which wasnt in devel

File: modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C (hunk 2 above)
Comment (GiudGiud) [reply modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:470]: ah my bad your assert was more restrictive. I ll add it to Rayleigh PR

**Pull request merged**